src/arrayMethodJoin.js

```diff
@@ -4,8 +4,28 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
+  [].__proto__.join2 = function(separator = ',') {
     // write code here
+    if (this.length === 0) {
+      return '';
+    }
+
+    let joinedExpression = '';
+
+    for (let i = 0; i < this.length - 1; i++) {
+      if (this[i] === null || this[i] === undefined) {
+        joinedExpression += separator;
+        continue;
+      }
+      joinedExpression += this[i];
+      joinedExpression += separator;
```

Comment:
you check if its null but even if it is you still add it after the is statement, I don't think it's intended